include/deal.II/numerics/nonlinear.h

```diff
@@ -490,6 +490,7 @@ template <typename VectorType>
 NonlinearSolverSelector<VectorType>::NonlinearSolverSelector(
   const AdditionalData &additional_data)
   : additional_data(additional_data)
+  , mpi_communicator(MPI_COMM_WORLD)
```

Comment:
No, this is not right. We should never assume that the user meant to use `MPI_COMM_WORLD` unless we are sure that this is a sequential program.

Reply:
I guess you mean MPI_COMM_SELF?

Reply:
@bangerth This is the constructor for the non-MPI variant. There is a second constructor that takes an `mpi_communicator` object.

I agree with @kronbichler, it should be `MPI_COMM_SELF` here. If the `NonlinearSolverSelector` is created without an MPI communicator then `MPI_COMM_SELF` is the right default value.

Reply:
```suggestion
NonlinearSolverSelector<VectorType>::NonlinearSolverSelector(
  const AdditionalData &additional_data)
  : additional_data(additional_data)
  , mpi_communicator(MPI_COMM_SELF)
```

Reply:
`MPI_COMM_SELF` works too in this case. Of course, either way you have the issue that someone may simply forget passing in the communicator, gets `MPI_COMM_SELF` here, and the program is going to deadlock in the next vector operation.

Could we check somewhere whether the communicator we set is the same as the one used by the initial guess vector used in the main function of this class?